hooks/useXRPStaking.ts

```
@@ -56,46 +56,46 @@ export function useXRPStaking(): StakingService {
         }
 
         // Get staker balance from Assets Precompile
-        const { success, stakerBalanceResponse } =
-          await getStakerBalanceByToken(
-            boundEvmAddress as `0x${string}`,
-            XRP_CHAIN_ID,
-            XRP_TOKEN_ADDRESS,
-          );
-
-        if (!success || !stakerBalanceResponse) {
+        const { data: stakerBalanceResponse } = useStakerBalanceByToken(
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)